adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/impl/Application.java

```diff
@@ -21,7 +21,9 @@
 /**
  * The Hono AMQP main application class.
  */
-@ComponentScan(basePackages = { "org.eclipse.hono.adapter.amqp", "org.eclipse.hono.service.metric" })
+@ComponentScan("org.eclipse.hono.adapter.amqp")
+@ComponentScan("org.eclipse.hono.adapter.amqp.impl")
+@ComponentScan("org.eclipse.hono.service.metric")
```

Comment:
I am not quite convinced that this is the same as specifying the packages in a single annotation. Do you have any source where the semantics of multiple instances of the same annotation on a type is defined, in particular in the Spring context?

Reply:
I think it is:

* https://docs.oracle.com/javase/tutorial/java/annotations/repeating.html
* https://github.com/spring-projects/spring-framework/blob/bc283ed7d70570bffba0f2a8c82b8e666a87d9f4/spring-context/src/main/java/org/springframework/context/annotation/ComponentScan.java#L58
* https://github.com/spring-projects/spring-framework/blob/4.3.x/spring-context/src/main/java/org/springframework/context/annotation/ComponentScans.java